### PR TITLE
[IMP] add --remove-old-repos options to oca-clone-everything

### PR DESCRIPTION
**Pull request by sbidoul**
This is a prerequisite to fully automating the oca-gen-addons-table runs.

### CODE REVIEW

**Comment by gurneyalex** [comment]
the script works with github if you are authenticated already. The hardcoded list should be completely removed IMO.

**Comment by pedrobaeza** [comment]
Let me work on this in the following days to avoid the need to clone everything

**Comment by sbidoul** [comment]
@gurneyalex 

> The hardcoded list should be completely removed IMO.

You mean the hardcoded list in https://github.com/OCA/maintainer-tools/blob/master/tools/oca_projects.py?

> the script works with github if you are authenticated already

Yes. Do you mean it can not work with a generic OCA bot user?

**Comment by sbidoul** [comment]
@pedrobaeza 

> Let me work on this in the following days to avoid the need to clone everything

Why? This script works and is well tested. Note the clone only occurs only the first time (then it's only fetching the changes).

**Comment by pedrobaeza** [comment]
But you need to keep that storage room for all the repositories. Have you seen migrate_branch.sh? I only get what's relevant from the repo. For example, the repo manifest: https://github.com/OCA/maintainer-tools/blob/master/tools/migrate_branch.py#L165, and then, making the commit via the GitHub API, not with local git.

**Comment by sbidoul** [comment]
@pedrobaeza I have seen migrate branch. I understand in some situations it's good to manipulate the github API directlly. In other situations it is either necessary to have a local clone of everything (for instance #156), or simply easier to works with with simple shell scripts.

Regarding this PR, is there an objection to have this new `--remove-old-repos` options?

**Comment by pedrobaeza** [comment]
:+1:

**Comment by gurneyalex** [comment]
@sbidoul yes that's the list I was refering to. It's a pain to maintain and the oca_projects.py only defaults to this list if you have not signed in github (using oca-github-login), otherwise the list is fetched from github. 

@pedrobaeza please don't break oca-clone-everything, we use it daily to keep a local mirror of all OCA projects and branches. And it is precious when you want to run mass commit scripts. 

:+1:

**Comment by pedrobaeza** [comment]
Why I'm going to break oca-clone-everything? That's not what I want. I just want to change this script to not need this previous step.

File: tools/clone_everything.py [hunk 1]
```diff
@@ -27,6 +28,12 @@ def clone(organization_remotes=None):
                 cmd = ['git', '--git-dir=' + os.path.join(project, '.git'),
                        'fetch', organization_remote]
                 subprocess.call(cmd)
+    if remove_old_repos:
+        for d in os.listdir('.'):
+            if d not in OCA_REPOSITORY_NAMES and \
+                    os.path.isdir(d) and \
+                    os.path.isdir(os.path.join(d, '.git')):
+                subprocess.check_call(['rm', '-fr', d])
```
Comment (gurneyalex) [inline tools/clone_everything.py]: `shutil.rmtree`?

**Comment by pedrobaeza** [comment]
I was referring the Bidoul's script that update the README's, which was the original reason to modify this

**Comment by gurneyalex** [comment]
@pedrobaeza ok. I would not bother (unless you have lots of time). This is going to run on one of the OCA's big servers :smile_cat:

**Pull request merged**